hier_config/platforms/fortigate_fortios/driver.py

```diff
@@ -0,0 +1,66 @@
+from collections.abc import Iterable
+from typing import Optional
+
+from hier_config.child import HConfigChild
+from hier_config.models import (
+    MatchRule,
+    ParentAllowsDuplicateChildRule,
+    SectionalExitingRule,
+)
+from hier_config.platforms.driver_base import HConfigDriverBase, HConfigDriverRules
+
+
+class HConfigDriverFortigateFortiOS(HConfigDriverBase):
+    """Driver for FortiGate OS."""
+
+    @staticmethod
+    def _instantiate_rules() -> HConfigDriverRules:
+        return HConfigDriverRules(
+            sectional_exiting=[
+                SectionalExitingRule(
+                    match_rules=(MatchRule(startswith="config"),), exit_text="end"
+                ),
+                SectionalExitingRule(
+                    match_rules=(
+                        MatchRule(startswith="config"),
+                        MatchRule(startswith="edit"),
+                    ),
+                    exit_text="next",
+                ),
+            ],
+            parent_allows_duplicate_child=[
+                ParentAllowsDuplicateChildRule(match_rules=()),
+            ],
+        )
+
+    def negate_with(self, config: HConfigChild) -> Optional[str]:
```

Comment:
You probably want to implement `swap_negation()` instead of `negate_with()` for this. 
see https://github.com/netdevops/hier_config/blob/2db52ec887b416b3e21c14d7d829cca3386abd6f/hier_config/platforms/vyos/driver.py#L7 and https://github.com/netdevops/hier_config/blob/2db52ec887b416b3e21c14d7d829cca3386abd6f/hier_config/platforms/juniper_junos/driver.py#L7